public/app/core/services/keybindingSrv.ts

```diff
@@ -256,6 +256,21 @@ export class KeybindingSrv {
       }
     });
 
+    // toggle all panel legends
+    this.bind('d l', () => {
+      const panels = dashboard.panels.filter(panel => {
```

Comment:
Could you move this to code DashboardModel and write a 2-3 unit tests for this code?

Reply:
No problem! I started this work but having some trouble because "[Webpack] TS2339: Property 'legend' does not exist on type 'PanelModel'."

I'm guessing because DashboardModels.panels is typed in DashboardModel, where as in keyBinding, it is not. How can I work around this?

Reply:
you can add an optional legend propety to PanelModel.  

legend?: {show: boolean};